### PR TITLE
Updated documentation for SRT_EPOLL_UPDATE

### PR DESCRIPTION
**Pull request by ethouris**

### CODE REVIEW

File: docs/API-functions.md [hunk 1]
```diff
@@ -2096,20 +2096,27 @@ Possible epoll flags are the following:
...
-socket that has the `SRTO_GROUPCONNECT` flag set to allow group connections.
-This event is intended for internal use only, and is triggered for group
-connections when a new link has been established for a group that is
-already connected (that is, has at least one connection established).
+The `SRT_EPOLL_IN`, `SRT_EPOLL_OUT` and `SRT_EPOLL_ERR` events are by default
+**level-triggered**. With `SRT_EPOLL_ET` flag they become **edge-triggered**.
+The `SRT_EPOLL_UPDATE` flag is always edge-triggered and it designates a
+special event that happens on a listener socket that has the
+`SRTO_GROUPCONNECT` flag set to allow group connections, or on a group. This
+is triggered in the following situations:
```
Comment (stevomatthews) [inline docs/API-functions.md:2110]:
```suggestion
The `SRT_EPOLL_IN`, `SRT_EPOLL_OUT` and `SRT_EPOLL_ERR` events are by default
**level-triggered**. With `SRT_EPOLL_ET` flag they become **edge-triggered**.

The `SRT_EPOLL_UPDATE` flag is always edge-triggered. It designates a
special event that happens on a group, or on a listener socket that has the
`SRTO_GROUPCONNECT` flag set to allow group connections. This flag
is triggered in the following situations:
```

File: docs/API-functions.md [hunk 2]
```diff
@@ -2096,20 +2096,27 @@ Possible epoll flags are the following:
...
+
+* for group connections, when a new link has been established for a group that
+is already connected (that is, has at least one connection established), reported
+for the listener socket that was accepting the connection. This is intended for
+internal use only. It can be understood as such that a reporting connection for
+the first time results in reporting the group connection on that listener, but
+when the group is already connected, this event is reported instead.
+
+* when one of group member connection has been broken
```
Comment (stevomatthews) [inline docs/API-functions.md:2119]:
```suggestion
* for group connections, when a new link has been established for a group that is already 
connected (that is, has at least one connection established), `SRT_EPOLL_UPDATE` is 
reported for the listener socket accepting the connection. This is intended for internal 
use only. An initial connection results in reporting the group connection on that listener. 
But when the group is already connected, `SRT_EPOLL_UPDATE`  is reported instead.

* when one of group member connection has been broken
```

**Review by stevomatthews — COMMENTED** [review]
Minor edits.

File: docs/API-functions.md [hunk 3]
```diff
@@ -2096,20 +2096,27 @@ Possible epoll flags are the following:
...
+
+* for group connections, when a new link has been established for a group that is already 
+connected (that is, has at least one connection established), `SRT_EPOLL_UPDATE` is 
+reported for the listener socket accepting the connection. This is intended for internal 
+use only. An initial connection results in reporting the group connection on that listener. 
+But when the group is already connected, `SRT_EPOLL_UPDATE`  is reported instead.
+
+* when one of group member connection has been broken
 
 Note that at this time the edge-triggered mode is supported only for SRT
```
Comment (jeandube) [inline docs/API-functions.md:2121]: Is this the line where we must deduce that srt_epoll_uwait() must be used for SRT_EPOLL_UPDATE and not srt_epoll_wait()?

Comment (ethouris) [reply docs/API-functions.md:2121]: No. Both these functions do exactly the same thing, except that *uwait can only work with eids subscribed for SRT sockets.

Comment (ethouris) [reply docs/API-functions.md:2121]: Damned. Theoretically yes, but looxlike another update is required.

**Pull request merged**